app/controllers/supervisor/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Supervisor::CommentsController < Supervisor::BaseController
+
+  def create
+    comment = Comment.new(comment_params)
+
+    if comment.save
+      CommentMailer.email(comment).deliver_later
```

Comment:
I would rather have a validation on comment, that it will only save correctly if text is present. Furthermore I prefer the mailing to remain in the controller, I prefer not to use callbacks on the model for cross-sectional stuff. This makes testing/seeding/... very hard.